File: app/src/main/java/jp/juggler/subwaytooter/api/TootTask.java
package jp.juggler.subwaytooter.api;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

public interface TootTask {

	// 非同期処理をここに実装する
	TootApiResult background( @NonNull TootApiClient client );

	// 非同期処理が終わったらメインスレッドから実行される
	// 処理がキャンセルされた場合、 resultはnullになるかもしれない
	void handleResult( @Nullable TootApiResult result );
	
}

File: app/src/main/java/jp/juggler/subwaytooter/api/TootTask.kt
package jp.juggler.subwaytooter.api

interface TootTask {
	
	// 非同期処理をここに実装する
	suspend fun background(client : TootApiClient) : TootApiResult?
	
	// 非同期処理が終わったらメインスレッドから実行される
	// 処理がキャンセルされた場合、 resultはnullになるかもしれない
	suspend fun handleResult(result : TootApiResult?)
}
